### DISCUSSION TITLE
Confusion about workflow with Chezmoi

### DISCUSSION THREAD

**Discussion by mafsi** [twpayne/chezmoi#2406] [Q&A] [answered]
In [Edit the source state](https://www.chezmoi.io/quick-start/#start-using-chezmoi-on-your-current-machine) from Quick Start, we find

```
$ chezmoi edit ~/.bashrc
```
as a daily routine to modify files that we added in `source directory`, and if we are satisfied, `Apply the changes: $ chezmoi -v apply`.

This _is unusual to me_ since I always used directly my editor to modify the file:  `$ vim ~/.bashrc`

### Here is my workflow:

1. `vim ~/.bashrc` & make usual changes
2. chezmoi re-add
3. chezmoi cd
4. git add `executable_dot_bashrc`
5. git commit -m "message here"
6. git push

**_Am I missing something? Am I doing the wrong way, or didn't I understand `chezmoi`?_**

**Comment by bradenhilton** [comment, marked as answer]
There's nothing wrong with that workflow if it works for you. Note that if you are using templates and edit the file in the destination directory, `chezmoi re-add` will not overwrite the templates.

Personally I navigate to the source directory in my editor (VSCode or Neovim), edit the files, then run `chezmoi diff` and finally `chezmoi apply` if I'm happy with the changes.
